pkg/scheduler/testing/wrappers.go

```diff
@@ -466,6 +466,21 @@ func (n *NodeWrapper) Label(k, v string) *NodeWrapper {
 	return n
 }
 
+// Taint applies a given {k, v} tuple as a taint to the inner node.
+// The prefer flag specifies if the taint is a preference(soft) or requirement(hard)
+// TODO: If needed expand the prefer option to include Execute later
+func (n *NodeWrapper) Taint(k, v string, prefer bool) *NodeWrapper {
```

Comment:
I'd prefer to change the signature to `Taint(k, v string, effect v1.TaintEffect)`